### PR TITLE
Viewer zoom changes

### PR DESCRIPTION
**Pull request by gestchild**
Addresses the functional changes of #4850, styling in a separate PR

### CODE REVIEW

File: common/utils/convert-image-uri.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ function wordPressImageTemplate(baseUrl: string) {
     template.expand(Object.assign({}, defaultOpts, opts));
 }
 
-type IiifUriOpts = {|
+export type IiifUriOpts = {|
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]: Any chance we could rename to `IIIFUriProps` to be consistent while we're here?

File: common/views/components/Header/Header.js [hunk 2]
```diff
@@ -34,88 +34,81 @@ const links = [
...
-          <div className="header__brand">
-            <a href="/" className="header__brand-link">
-              <WellcomeCollectionBlack />
-            </a>
-          </div>
-          <nav
-            id="header-nav"
-            className="header__nav"
-            aria-labelledby="header-burger-trigger"
+const Header = withToggler(({ siteSection, toggle, isActive }: Props) => (
```
Comment (jamesgorrie) [inline common/views/components/Header/Header.js]: We've removed withToggler elsewhere if we'd like to remove it here.
Bikeshedding though.

File: common/views/components/ImageViewer/ImageViewer.js [hunk 3]
```diff
@@ -68,207 +84,112 @@ const ImageWrapper = styled.div`
...
+          });
+
+          return urlString && `${urlString} ${width}w`;
+        })
+        .join(',')
+  );
+  const [rotation, setRotation] = useState(0);
+
+  useEffect(() => {
+    urlTemplate &&
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]: This `&&` logic has caused quite a lof of confustion while working through other bits of the code. It's _OK_ for react as an if as it's all you've got really, but could we try use more explanatory logic out of the templates?

```js
if (urlTemplate) {
 //... blah
}
```

This does raise question as to why we need the conditional as the `urlTemplate` isn't optional?

File: common/views/components/ImageViewer/ImageViewer.js [hunk 4]
```diff
@@ -68,207 +84,112 @@ const ImageWrapper = styled.div`
...
+      );
+  }, [infoUrl, rotation]);
 
   function routeChangeStart(url: string) {
     if (window.history.state.as !== url) {
       setImageLoading(true);
     }
   }
+
+  const escapeCloseViewer = ({ keyCode }: KeyboardEvent) => {
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]: Would this make sense be a `onKeyDown`  or `keyboardShortcutsListener` to make sure we don't proliferate the listeners?

File: common/views/components/ImageViewer/ImageViewer.js [hunk 5]
```diff
@@ -68,207 +84,112 @@ const ImageWrapper = styled.div`
...
-  src: string,
-  srcSet: string,
   width: number,
   height?: number,
   infoUrl: string,
   lang: ?string,
+  alt: string,
   tabbableControls: boolean,
+  urlTemplate: IiifUriOpts => string,
+  presentationOnly?: boolean,
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]: [We never seem to pass this prop through](https://github.com/wellcometrust/wellcomecollection.org/pull/4906/files#diff-73c6a0583e43bee87b93420f87e90ce6R57-R65) - so where are we using it?

File: common/views/components/ZoomedImage/ZoomedImage.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+const ZoomedImage = ({ id, infoUrl, setShowViewer }: Props) => {
+  const [scriptError, setScriptError] = useState(false);
+  const [viewer, setViewer] = useState(null);
+  const zoomStep = 0.5;
+  const firstControl = useRef(null);
+  const lastControl = useRef(null);
+  const zoomedImage = useRef(null);
+  const handleScriptError = error => {
+    Raven.captureException(new Error(`OpenSeadragon error: ${error}`), {
```
Comment (jamesgorrie) [inline common/views/components/ZoomedImage/ZoomedImage.js]: Looking at how this has been applied - it's not an OSD error - but rather a fetching of the image error?

File: common/views/components/ZoomedImage/ZoomedImage.js [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                  width: 400,
+                },
+              ],
+            },
+          ],
+        });
+        setViewer(osdViewer);
+      })
+      .catch(error => {
+        handleScriptError(error);
```
Comment (jamesgorrie) [inline common/views/components/ZoomedImage/ZoomedImage.js]: Looking at how this has been applied - it's not an OSD error - but rather a fetching of the image error?

File: common/views/components/ZoomedImage/ZoomedImage.js [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+      });
+  }
+
+  useEffect(() => {
+    setViewer(setupViewer(infoUrl, id, handleScriptError));
```
Comment (jamesgorrie) [inline common/views/components/ZoomedImage/ZoomedImage.js]: This function is past around like crazy - should we just set it where it's used?

File: common/views/components/ImageViewer/ImageViewer.js (hunk 5 above)
Comment (gestchild) [reply common/views/components/ImageViewer/ImageViewer.js]: https://github.com/wellcometrust/wellcomecollection.org/blob/035fb847022ddefcd702800a5d5b50ce3d960e4a/common/views/components/IIIFViewer/IIIFViewer.js#L803

File: common/utils/convert-image-uri.js [hunk 9]
```diff
@@ -81,7 +81,7 @@ function wordPressImageTemplate(baseUrl: string) {
     template.expand(Object.assign({}, defaultOpts, opts));
 }
 
-type IiifUriOpts = {|
+export type IiifUriProps = {|
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]:
```suggestion
export type IIIFUriProps = {|
```

File: common/utils/convert-image-uri.js [hunk 10]
```diff
@@ -100,7 +100,7 @@ export function iiifImageTemplate(infoJsonLocation: string) {
     format: 'jpg',
   };
   const template = urlTemplate.parse(templateString);
-  return (opts: IiifUriOpts) =>
+  return (opts: IiifUriProps) =>
```
Comment (jamesgorrie) [inline common/utils/convert-image-uri.js]:
```suggestion
  return (opts: IIIFUriProps) =>
```

File: common/views/components/ImageViewer/ImageViewer.js [hunk 11]
```diff
@@ -1,14 +1,31 @@
 // @flow
 import { useState, useEffect } from 'react';
 import Router from 'next/router';
+import dynamic from 'next/dynamic';
 import styled from 'styled-components';
-import fetch from 'isomorphic-unfetch';
-import Raven from 'raven-js';
-import { trackEvent } from '../../../utils/ga';
+import { type IiifUriProps } from '@weco/common/utils/convert-image-uri';
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]:
```suggestion
import { type IIIFUriProps } from '@weco/common/utils/convert-image-uri';
```

File: common/views/components/ImageViewer/ImageViewer.js [hunk 12]
```diff
@@ -68,207 +84,107 @@ const ImageWrapper = styled.div`
...
   id: string,
-  src: string,
-  srcSet: string,
   width: number,
   height?: number,
   infoUrl: string,
   lang: ?string,
+  alt: string,
   tabbableControls: boolean,
+  urlTemplate: IiifUriProps => string,
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewer.js]:
```suggestion
  urlTemplate: IIIFUriProps => string,
```

File: common/views/components/ZoomedImage/ZoomedImage.js [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+  const handleScriptError = error => {
+    Raven.captureException(new Error(`OpenSeadragon error: ${error}`), {
+      tags: {
+        service: 'dlcs',
+      },
+    });
+    setScriptError(true);
+  };
+
+  function setupViewer(imageInfoSrc, viewerId, handleScriptError) {
```
Comment (jamesgorrie) [inline common/views/components/ZoomedImage/ZoomedImage.js]: still unsure as to why `handleScriptError` is an parameter when it's always the same?

**Pull request merged**